consent_manager/api.rst

```diff
@@ -0,0 +1,606 @@
+After successful, Consent Manager internally send consent settings to tracking server and force page view on tags.
+
+Code::
+
+    ppms.cm.api('setComplianceSettings', settings, onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:setComplianceSettings', parameters: [settings, onFulfilled, onRejected]});
+
+.. object:: settings
+
+    The consent settings object.
+
+        Example::
+
+            {consents: {analytics: {status: -1}}}
+
+    Where `consent.analytics` is consent type and status indicate:
+
+    * `-1` - user does not interact, e.q. close consent popup without any decision
+    * `0` - user reject consent
+    * `1` - user approve consent
+
+.. function:: onFulfilled()
+
+    The fulfilment handler callback.
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code).
+
+    :param string|object error: **Required** Error code or exception.
+
+Get compliance settings
+`````````````````````````````````````
+Return current privacy settings. Might be useful for initializing custom decision view.
+When there is no decisions, just return empty object. This state can be used for detect first time user interaction with consent mechanism.
+
+Code::
+
+    ppms.cm.api('getComplianceSettings', onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:getComplianceSettings', parameters: [onFulfilled, onRejected]});
+
+.. object:: settings
+
+    The consent settings object.
+
+        Example::
+
+            {consents: {analytics: {status: -1, updatedAt: '2018-07-03T12:18:19.957Z'}}}
+
+    Where `consent.analytics` is consent type and status indicate:
+
+    * `-1` - user does not interact, e.q. close consent popup without any decision
+    * `0` - user reject consent
+    * `1` - user approve consent
+
+.. function:: onFulfilled(settings)
+
+    The fulfilment handler callback (called with result).
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code).
+
+    :param string|object error: **Required** Error code or exception.
+
+Send data request
+`````````````````````````````````````
+Command send data subject request to Consent Manager collector.
+
+Code::
+
+    ppms.cm.api('sendDataRequest', request, onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:sendDataRequest', parameters: [request, onFulfilled, onRejected]});
+
+.. object:: request
+
+    The subject data request.
+
+        Example::
+
+            {content: '', email: '', type: 'change_data|view_data|delete_data'}
+
+.. function:: onFulfilled()
+
+    The fulfilment handler callback.
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code).
+
+    :param string|object error: **Required** Error code or exception.
+
+Example usage
+--------
+Based on above listed commands there are many possibilities to implement custom consent gathering. Below it is listed
+simple implementation using jquery.
+
+.. code-block:: html
```

Comment:
can we use http://www.sphinx-doc.org/en/stable/markup/code.html#directive-literalinclude to include this from separate file ?